lib/new.js

```diff
@@ -0,0 +1,687 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require('path');
+const puppeteer = require('puppeteer-core');const findChrome = require('./find_chrome');
+const {rpc} = require('../rpc');
+const debugApp = require('debug')('carlo:app');
+const debugServer = require('debug')('carlo:server');
+const {Color} = require('./color');
+const {HttpRequest} = require('./http_request');
+
+const fs = require('fs');
+const util = require('util');
+const {URL} = require('url');
+const EventEmitter = require('events');
+const fsReadFile = util.promisify(fs.readFile);
+
+let testMode = false;
```

Comment:
format: style mismatch:
```javascript
32|
33|let testMode = false;
34|
```
```suggestion
let testMode = false;class App extends EventEmitter {/**
```

⏎⏎ at column 22 should be removed.
Triggered rule # 143
```
  +1.roles in {MAP} and not in {ARGUMENT}
	∧ +2.roles not in {RIGHT}
	∧ ^1.roles not in {DECLARATION}
	∧ ^2.roles not in {BINARY, MAP}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved = , and not in {:, .}
	∧ -2.diff_line = 0
	∧ -3.roles not in {VALUE}
	∧ -4.roles in {LITERAL}
	⇒ y = ⏎
	Confidence: 0.812. Support: 8.
```
⏎␣⁺␣⁺ at column 33 should be removed.
Triggered rule # 121
```
  +1.roles in {ARGUMENT}
	∧ +2.roles not in {RIGHT}
	∧ +4.reserved = const
	∧ ^1.roles not in {DECLARATION}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {:, .}
	∧ -2.diff_line = 0
	∧ -3.length ≤ 8
	∧ -3.diff_offset ≥ 4
	∧ -3.label in {<space>}
	⇒ y = '
	Confidence: 0.976. Support: 21.
```